### PR TITLE
fix: fetch `scriptPubKeys` for deposit transaction inputs

### PR DESCRIPTION
**Pull request by djordon**
## Description

Closes https://github.com/stacks-network/sbtc/issues/674

## Changes

* Fetch the transaction info for deposits, and store the prevout `scriptPubKey`s.
* Do not take note of deposit requests that have not confirmed. Before we would keep them around in memory if they were in the mempool and passed validation.
* Added more helpful error messages to our error variants.

## Checklist:

- [x] I have performed a self-review of my code

### CODE REVIEW

File: signer/src/testing/block_observer.rs [hunk 1]
```diff
@@ -60,13 +61,29 @@ impl TestHarness {
...
     /// Add a single deposit transaction to the test harness.
     pub fn add_deposit(&mut self, txid: Txid, response: GetTxResponse) {
-        self.deposits.insert(txid, response);
+        let tx_info = BitcoinTxInfo {
+            in_active_chain: response.block_hash.is_some(),
+            fee: bitcoin::Amount::from_sat(1000),
+            tx: response.tx.clone(),
+            txid: response.tx.compute_txid(),
+            hash: response.tx.compute_wtxid(),
+            size: response.tx.vsize() as u64,
```
Comment (matteojug) [inline signer/src/testing/block_observer.rs:76]: nit: (maybe?)
```suggestion
            size: response.tx.total_size() as u64,
```

File: signer/src/storage/model.rs [hunk 2]
```diff
@@ -81,11 +81,12 @@ pub struct DepositRequest {
 
 impl From<Deposit> for DepositRequest {
     fn from(deposit: Deposit) -> Self {
-        let tx_input_iter = deposit.tx.input.into_iter();
+        let tx_input_iter = deposit.tx_info.vin.into_iter();
```
Comment (matteojug) [inline signer/src/storage/model.rs:84]: Is there a reason to prefer `tx_info.vin` to `tx_info.tx.input`? Note that in the test below, in `add_deposit`, we are not populating `vin`.

File: signer/tests/integration/transaction_signer.rs [hunk 3]
```diff
@@ -275,3 +278,84 @@ async fn get_signer_public_keys_and_aggregate_key_falls_back() {
...
+    let mut requests = db
+        .get_pending_deposit_requests(&chain_tip, 100)
+        .await
+        .unwrap();
+    // There should only be the one deposit request that we just fetched.
+    assert_eq!(requests.len(), 1);
+    let request = requests.pop().unwrap();
+
+    let network = InMemoryNetwork::new();
+    let mut coord = TxSignerEventLoop {
```
Comment (matteojug) [inline signer/tests/integration/transaction_signer.rs:325]: nit:
```suggestion
    let mut signer = TxSignerEventLoop {
```

File: signer/src/storage/model.rs (hunk 2 above)
Comment (djordon) [reply signer/src/storage/model.rs:84]: > Is there a reason to prefer `tx_info.vin` to `tx_info.tx.input`? 

Yes, `tx_info.tx.input` is the "raw" transaction's input, as stored on a bitcoin block. The raw inputs do not include the `scriptPubKey` of the inputs, which is what we want here.

> Note that in the test below, in `add_deposit`, we are not populating `vin`.

Yeah, it wasn't necessary for those tests and it's non-trivial to populate correctly.

**Review by matteojug — APPROVED** [review]

File: signer/tests/integration/transaction_signer.rs (hunk 3 above)
Comment (djordon) [reply signer/tests/integration/transaction_signer.rs]: Done in https://github.com/stacks-network/sbtc/pull/675/commits/0902b2a9b7a79745473e7b90405ef5e4860113d7.

File: signer/src/testing/block_observer.rs (hunk 1 above)
Comment (djordon) [reply signer/src/testing/block_observer.rs]: Done in https://github.com/stacks-network/sbtc/pull/675/commits/0902b2a9b7a79745473e7b90405ef5e4860113d7.

File: signer/src/error.rs [hunk 4]
```diff
@@ -287,7 +287,7 @@ pub enum Error {
     StacksNodeRequest(#[source] reqwest::Error),
 
     /// We failed to submit the transaction to the mempool.
-    #[error("{0}")]
+    #[error("transaction rejected: {0}")]
```
[the remark this answers is not in the document]
Comment (djordon) [reply signer/src/error.rs]: https://github.com/stacks-network/sbtc/pull/675/commits/0902b2a9b7a79745473e7b90405ef5e4860113d7

**Pull request merged**